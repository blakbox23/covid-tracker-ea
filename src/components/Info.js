import React from 'react';
import { VscGithub } from 'react-icons/vsc';
import { AiOutlineLinkedin } from 'react-icons/ai';
import Nav from './Nav';

function Info() {
  return (
    <div className="info-comp">
      <Nav />
      <p className="title">
        SOURCES
      </p>
      <div className="stats-list">
        <div className="flex stats-list-item">
          <span><a target="blank" href="https://www.worldometers.info/coronavirus/">Worldometers</a></span>
        </div>
        <div className="flex stats-list-item">
          <span><a target="blank" href="https://github.com/disease-sh/API">Disease-sh API</a></span>
        </div>
      </div>
      <p className="title">
        DEVELOPER
      </p>
      <div className="stats-list">
        <div className="flex stats-list-item">
          <span>
            {' '}
            <VscGithub />
            {' '}
            <span><a target="blank" href="https://github.com/blakbox23/">blakbox23</a></span>
          </span>
        </div>
        <div className="flex stats-list-item">
          <span>
            {' '}
            <AiOutlineLinkedin />
            {' '}
            <span><a target="blank" href="https://www.linkedin.com/in/peter-mbuthia/">blakbox23</a></span>
          </span>
        </div>
      </div>
    </div>
  );
}

export default Info;
