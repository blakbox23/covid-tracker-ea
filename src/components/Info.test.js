import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Info from './Info';

describe('Info page snapshot', () => {
    test('renders correct snapshot', () => {
      const tree = renderer
        .create(        
        <Router>
            <Info />
            </Router>)

        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  test('renders Info component', () => {
    render(
        <Router>
          <Info />
        </Router>,
      )
    expect(screen.getByText('Info')).toBeInTheDocument();
    expect(screen.getByText('GENERAL')).toBeInTheDocument();
    expect(screen.getByText('SOURCES')).toBeInTheDocument();
    expect(screen.getByText('DEVELOPER')).toBeInTheDocument();
  });
