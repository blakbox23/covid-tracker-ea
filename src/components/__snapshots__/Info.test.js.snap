// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info page snapshot renders correct snapshot 1`] = `
<div
  className="info-comp"
>
  <div
    className="navbar flex"
  >
    <div>
      <a
        className="back-btn"
        href="/"
        onClick={[Function]}
      >
        &lt;
      </a>
    </div>
    <div
      className="logo-txt"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Covid in Africa
      </a>
    </div>
    <a
      className="logo"
      href="/info"
      onClick={[Function]}
    >
      <div>
        <svg
          className="src-logo"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zM11 7h2v2h-2V7zm0 4h2v6h-2v-6z"
            />
          </g>
        </svg>
      </div>
    </a>
  </div>
  <header
    className="header flex"
  >
    <div
      className="header-img"
    />
    <div
      className="header-text"
    >
      <div>
        <b>
          Info
        </b>
      </div>
      More about the app
    </div>
  </header>
  <p
    className="title"
  >
    GENERAL
  </p>
  <div
    className="stats-list"
  >
    <div
      className="flex stats-list-item"
    >
      This is published data on the spread of the Covid 19 in Africa.
    </div>
    <div
      className="flex stats-list-item"
    >
      The API used by this App is updated everyday.
    </div>
  </div>
  <p
    className="title"
  >
    SOURCES
  </p>
  <div
    className="stats-list"
  >
    <div
      className="flex stats-list-item"
    >
      <span>
        <a
          href="https://www.worldometers.info/coronavirus/"
          target="blank"
        >
          Worldometers
        </a>
      </span>
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
        <a
          href="https://github.com/disease-sh/API"
          target="blank"
        >
          Disease-sh API
        </a>
      </span>
    </div>
  </div>
  <p
    className="title"
  >
    DEVELOPER
  </p>
  <div
    className="stats-list"
  >
    <div
      className="flex stats-list-item"
    >
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm3.163 21.783h-.093a.513.513 0 0 1-.382-.14.513.513 0 0 1-.14-.372v-1.406c.006-.467.01-.94.01-1.416a3.693 3.693 0 0 0-.151-1.028 1.832 1.832 0 0 0-.542-.875 8.014 8.014 0 0 0 2.038-.471 4.051 4.051 0 0 0 1.466-.964c.407-.427.71-.943.885-1.506a6.77 6.77 0 0 0 .3-2.13 4.138 4.138 0 0 0-.26-1.476 3.892 3.892 0 0 0-.795-1.284 2.81 2.81 0 0 0 .162-.582c.033-.2.05-.402.05-.604 0-.26-.03-.52-.09-.773a5.309 5.309 0 0 0-.221-.763.293.293 0 0 0-.111-.02h-.11c-.23.002-.456.04-.674.111a5.34 5.34 0 0 0-.703.26 6.503 6.503 0 0 0-.661.343c-.215.127-.405.249-.573.362a9.578 9.578 0 0 0-5.143 0 13.507 13.507 0 0 0-.572-.362 6.022 6.022 0 0 0-.672-.342 4.516 4.516 0 0 0-.705-.261 2.203 2.203 0 0 0-.662-.111h-.11a.29.29 0 0 0-.11.02 5.844 5.844 0 0 0-.23.763c-.054.254-.08.513-.081.773 0 .202.017.404.051.604.033.199.086.394.16.582A3.888 3.888 0 0 0 5.702 10a4.142 4.142 0 0 0-.263 1.476 6.871 6.871 0 0 0 .292 2.12c.181.563.483 1.08.884 1.516.415.422.915.75 1.466.964.653.25 1.337.41 2.033.476a1.828 1.828 0 0 0-.452.633 2.99 2.99 0 0 0-.2.744 2.754 2.754 0 0 1-1.175.27 1.788 1.788 0 0 1-1.065-.3 2.904 2.904 0 0 1-.752-.824 3.1 3.1 0 0 0-.292-.382 2.693 2.693 0 0 0-.372-.343 1.841 1.841 0 0 0-.432-.24 1.2 1.2 0 0 0-.481-.101c-.04.001-.08.005-.12.01a.649.649 0 0 0-.162.02.408.408 0 0 0-.13.06.116.116 0 0 0-.06.1.33.33 0 0 0 .14.242c.093.074.17.131.232.171l.03.021c.133.103.261.214.382.333.112.098.213.209.3.33.09.119.168.246.231.381.073.134.15.288.231.463.188.474.522.875.954 1.145.453.243.961.364 1.476.351.174 0 .349-.01.522-.03.172-.028.343-.057.515-.091v1.743a.5.5 0 0 1-.533.521h-.062a10.286 10.286 0 1 1 6.324 0v.005z"
          />
        </svg>
         
        <span>
          <a
            href="https://github.com/blakbox23/"
            target="blank"
          >
            blakbox23
          </a>
        </span>
      </span>
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
          />
        </svg>
         
        <span>
          <a
            href="https://www.linkedin.com/in/peter-mbuthia/"
            target="blank"
          >
            blakbox23
          </a>
        </span>
      </span>
    </div>
  </div>
</div>
`;
